src/calculator/index.ts

```diff
@@ -321,24 +323,27 @@ export default class Calculator {
 
         /*TODO: take into account round matching token decimals instead of default 18 */
         const amountRoundToken =
-          (Number(amountUSD) / conversionRateRoundToken.price) *
-          Math.pow(10, 18);
+          (Number(amountUSD) / conversionRateRoundToken.price) * 10e18;
```

Comment:
you can use `.toLocaleString("fullwide", { useGrouping: false })` for that. 
otherwise we should convert the numbers before the multiplication that make them big in case they become too big for numbers